### ISSUE TITLE
Some questions about mAP evaluation

### ISSUE THREAD

**Issue by VG-Huai** [open-mmlab/mmrotate#76]
Hello, I got confused about the mAP implementation in mmrotate. It seems that the fuction `eval_map()` hasn't changed the confidence score threshold to get different precision and recall values. Instead, it gets precision and recall values according to the number of dets. 

https://github.com/open-mmlab/mmrotate/blob/ede3833bb638f630892b5f73ad49cbf3d8ee07b1/mmrotate/datasets/dota.py#L334

Actually, I've got a really high mAP value this way in my custom dataset, but the result doesn't match the visualization. Every class have really high ap but still have some wrong cases in visualization.

![middle_img_v2_6888ea78-8f4e-4cac-9f9a-4b0d329b7eag](https://user-images.githubusercontent.com/56375896/156999692-77110bb4-3122-42c7-8280-0874f61c768e.png)

**Comment by yangxue0827** [comment]
![image](https://user-images.githubusercontent.com/29257168/157001579-9f7fa5f3-9248-40c8-ab4d-9e4b3a6c8583.png)
Please confirm which part of the dataset you are evaluating. By default, DOTA dataset uses the same data as the training set for evaluation, so its results will be high. If this is the case for you, please modify the path to the validation dataset.

**Comment by VG-Huai** [comment]
We confirm that we only use training set for training, and evaluation set and test set for testing. Actually, we have a really big training set.
![image](https://user-images.githubusercontent.com/56375896/157003015-7fc9b9cb-2fb5-4d4d-801a-32c6bc0c697c.png)

**Comment by yangxue0827** [comment]
The calculation of ap does not need to set a threshold. In fact, ap is an overall indicator used to measure the area of the entire pr curve. What you said about using a threshold to calculate r and p probably refers to f-feature, which is a point on the pr curve and is often used to measure the effectiveness of visual visualizations.

**Comment by VG-Huai** [comment]
I researched a lot of articles about mAP evaluation, they got different precision and recall values by changing the confidence score threshold. It looks likes the figure below:
![middle_img_v2_120946da-2315-4ca1-850a-319dfb89ccdg](https://user-images.githubusercontent.com/56375896/157151892-74d2bdb4-75ad-416c-a57d-62bf3b35bc74.gif)

Here's the pseudo code:
![image](https://user-images.githubusercontent.com/56375896/157152167-b37b2090-d239-4915-884f-a729592230dd.png)

**Comment by jbwang1997** [comment]
The process of decreasing the score threshold can directly be finished by [np.cumsum](https://github.com/open-mmlab/mmrotate/blob/main/mmrotate/datasets/dota.py#L415-L420).

**Comment by VG-Huai** [comment]
Thanks a lot, I get it.

**Comment by CrapbagMo** [comment]
> 

hi, I seem to have encountered the same problem. Have you solved it yet?

**Comment by 19990101lrk** [comment]
> I researched a lot of articles about mAP evaluation, they got different precision and recall values by changing the confidence score threshold. It looks likes the figure below: ![middle_img_v2_120946da-2315-4ca1-850a-319dfb89ccdg](https://user-images.githubusercontent.com/56375896/157151892-74d2bdb4-75ad-416c-a57d-62bf3b35bc74.gif) [ ![middle_img_v2_120946da-2315-4ca1-850a-319dfb89ccdg](https://user-images.githubusercontent.com/56375896/157151892-74d2bdb4-75ad-416c-a57d-62bf3b35bc74.gif) ](https://user-images.githubusercontent.com/56375896/157151892-74d2bdb4-75ad-416c-a57d-62bf3b35bc74.gif) [ ](https://user-images.githubusercontent.com/56375896/157151892-74d2bdb4-75ad-416c-a57d-62bf3b35bc74.gif)
> 
> Here's the pseudo code: ![image](https://user-images.githubusercontent.com/56375896/157152167-b37b2090-d239-4915-884f-a729592230dd.png)

Hello, I would like to ask how to draw a PR curve based on test data in mmrotate

**Comment by Petros626** [comment]
@19990101lrk it's not supported by default or?
